Update getting-started.md

ihelmer07 opened the pull request:
## What? (description)

Clarifies even further that Talos is loaded to RAM until configuration is applied. Really trying to hammer it home. see #10349 

## Why? (reasoning)

Talos loading to RAM first is a unique aspect, outside the typical. Really hammering it home that if one restarts the machine, the OS is 'gone' and needs to be restarted. The reason https://github.com/siderolabs/talos/issues/10349 was because I went in with the assumption that the ISO wasn't needed any more before applying a configuration, which got me into the loop of UEFI shell.

rothgar commented on website/content/v1.9/introduction/getting-started.md:
```diff
@@ -84,16 +85,18 @@ The endpoint should be formatted like:
-> A configuration can be passed in on boot via kernel parameters or metadata servers.
-> See [Production Notes]({{< relref "prodnotes#configure-talos" >}}).
+enters maintenance mode and waits for a configuration to be provided.  
+> NOTE: Talos initially loads the OS to RAM, and only installs to disk after the configuration is applied. Removing the ISO and
```
I don't think your finished the sentence here.

ihelmer07 replied:
You're right - embarrassing. Fixed.

smira commented on website/content/v1.9/introduction/getting-started.md:
```diff
@@ -54,7 +54,8 @@ The latest ISO image can be found on the Github [Releases](https://github.com/si
 When booted from the ISO, Talos will run in RAM and will not install to disk until provided a configuration.
-Thus, it is safe to boot any machine from the ISO.
+Thus, it is safe to boot any machine from the ISO. 
+This also means restarting the machine without applying a configuration will restart the process. 
```
this seems to repeat the  sentence at line 56 in different words

smira commented on website/content/v1.9/introduction/getting-started.md:
```diff
@@ -84,16 +85,18 @@ The endpoint should be formatted like:
-> A configuration can be passed in on boot via kernel parameters or metadata servers.
-> See [Production Notes]({{< relref "prodnotes#configure-talos" >}}).
+enters maintenance mode and waits for a configuration to be provided.  
+> NOTE: Talos initially loads the OS to RAM, and only installs to disk after the configuration is applied. Removing the ISO and restarting the machine requires restarting the process completely.
```
I think it should be reworded, as Talos can be booted via a disk image. 

I don't see why restart is _that_ big of a problem. If the machine was ISO-booted, or PXE-booted, it's natural that on reboot it will boot once again from the same media.

If the machine was booted from a disk image, this note doesn't apply, but it confuses even more.

ihelmer07 replied:
reworded.

smira commented:
To be fair, I can't remember any OS that if booted from an ISO would install itself without any interaction from the user, so I don't see how Talos is unique here

ihelmer07 commented:
I hear with what you're saying.
My admittedly relatively inexperienced perspective is that a 'usual' USB image install process is:
1. Launch iso from usb.
2. Follow prompts if any.
3. Installed & OS displays

Many OS have restart protections 

Talos is:
1. Launch from ISO.
2. No prompts but lots of "installing"
3. Talos Summary page displays. <-- assumed it was 'done'
4. Apply config
5. Talos actually installed

The point is Talos has extra steps - Talos Step 3 is not the same as "usual" step 3. 
My usual 'how-to-install' assumptions were subverted with Talos which is the point caused the confusion. 

I'm only trying to prevent similar situation and make the documentation the best it can be.
Happy to remove/change as needed.

ihelmer07 commented:
@smira - Your wording is better. I've updated accordingly and removed line 58 as well.

Good to hear that 1.11 will make things more straightforward.

smira reviewed (approved):
thank you!

smira commented:
/m
